diff -ur rc-inetd-0.16.orig/rc-inetd rc-inetd-0.16/rc-inetd
--- rc-inetd-0.16.orig/rc-inetd	2006-04-10 15:46:49.000000000 +0200
+++ rc-inetd-0.16/rc-inetd	2006-04-10 15:58:58.028005000 +0200
@@ -12,10 +12,14 @@
 
 . /etc/rc.d/init.d/functions
 
+AUTO_CONFIG=yes
+
 # sanity check for /etc/sysconfig/rc-inetd.conf
-# Note: defaults must be read for each service, because they will be replaced
+# Note: defaults must also be read for each service, because they will be replaced
 #	by config of each service processed.
-if [ ! -f /etc/sysconfig/rc-inetd.conf ]; then
+if [ -f /etc/sysconfig/rc-inetd.conf ]; then
+	. /etc/sysconfig/rc-inetd.conf
+else
 	nls "Error: %s not found" /etc/sysconfig/rc-inetd.conf
 	nls " Inet server can't be run."
 	exit 6
@@ -147,12 +151,17 @@
 		msg_already_running Inet
 	fi
 
-	show "$(nls "Generating %s for %s" "$CONFIG_FILE" "$PROCESS_NAME")"
-	busy
-	pre_start_services
-	gen_config_file
-	DIDPARSE=$?
-	deltext;ok
+	if [ "$AUTO_CONFIG" != "no" ]; then
+		show "$(nls "Generating %s for %s" "$CONFIG_FILE" "$PROCESS_NAME")"
+		busy
+		pre_start_services
+		gen_config_file
+		DIDPARSE=$?
+		deltext;ok
+	else
+		DIDPARSE=1
+	fi
+
 	if [ "$DIDPARSE" -gt 0 ]; then
 		msg_starting "$PROCESS_NAME"
 		daemon $INETDAEMON $INETDAEMON_ARGS
@@ -180,10 +189,13 @@
 		exit 7
 	fi
 
-	show "$(nls "Generating %s for %s" "$CONFIG_FILE" "$PROCESS_NAME")"
-	busy
-	gen_config_file
-	deltext; ok
+	if [ "$AUTO_CONFIG" != "no" ]; then
+		show "$(nls "Generating %s for %s" "$CONFIG_FILE" "$PROCESS_NAME")"
+		busy
+		gen_config_file
+		deltext; ok
+	fi
+
 	show "$(nls "Reload %s service configuration" "$PROCESS_NAME")"
 	busy
 	reload_config
diff -ur rc-inetd-0.16.orig/rc-inetd.conf rc-inetd-0.16/rc-inetd.conf
--- rc-inetd-0.16.orig/rc-inetd.conf	2006-04-10 15:51:06.726550500 +0200
+++ rc-inetd-0.16/rc-inetd.conf	2006-04-10 15:51:49.989254250 +0200
@@ -1,5 +1,11 @@
 # This file contains several rc-inetd configuration parameters
 
+# Do you want rc-inetd to automagically generate inetd config?
+# Warning: this will overwrite current configuration without asking!
+# Default: yes
+
+AUTO_CONFIG="yes"
+
 # List deny services. List must contain services names separated by space
 # or tab.
 
